### ISSUE TITLE
Initial run on big table seems to make no progress

### ISSUE THREAD

**Issue by Jelle7** [toluaina/pgsync#87]
PGSync version: 1.1.28

Postgres version: 12.x

Elasticsearch version: 7.11.2

Redis version: 4.0.8

Python version: 3.8

Problem Description: 
After running bootstrap without issues,
I am running pgsync for the first time on a table with 72M existing rows (fetching a single child as well), the program seems to be struck at the first query (shown in verbose mode):
`Query:
SELECT JSON_BUILD_ARRAY(anon_1._keys) AS "JSON_BUILD_ARRAY_1",
...`

Is it trying to download the whole table as a json array? I've set the `QUERY_CHUNK_SIZE` to 1K, so I would assume it processes the data in batches of 1K. However after letting it run for hours on end, no records have been inserted in elasticsearch. 

When I run the program for smaller tables it does seem to work so my RDS settings seem to be correct. Is there any setting I can tweak to see any progress?

**Comment by toluaina** [comment]
- 72M is a lot of records.
- pgsync still syncs them in batches. 
- is this a flat table with no relationships and what is the structure of your schema?
- also do you have wide tables?
- this points to memory
- try reducing QUERY_CHUNK_SIZE even furthere and 
- this is an area I admit can be improved
- For this you will need to do some db [tuning](https://wiki.postgresql.org/wiki/Tuning_Your_PostgreSQL_Server) 
- you can also try increasing your `work_mem`

**Comment by Jelle7** [comment]
@toluaina This is not a flat table, denormalized using a single child in the schema. I pulled down the project and made some edits. 

I found that removing the `rowcount` in `_sync -> sync.py` and adding `result = conn.execution_options(stream_results=True).execute(query)` in the `query_yield -> base.py` function to stream the results worked for me. 

Sqlalchemy is now pulling down the data more efficiently in chunks and processing them to ES while I can follow the progress.
